web/tests/message_list_view.test.js

```diff
@@ -786,44 +787,77 @@ test("render_windows", ({mock_template}) => {
         assert.equal(view._render_win_end, end);
     }
 
+    function verify_move_and_no_move_range(move_target, opts = {}) {
+        // When we move to position X, we expect 400/2 = 200 messages on
+        // either side, unless that goes outside the `count`, in which
+        // case we'll specif it in `opts`.
+        const move_start = opts.move_start ?? move_target - 200;
+        const move_end = opts.move_end ?? move_target + 200;
+        verify_move(move_target, [move_start, move_end]);
+        // the no-move range is a 50 buffer on each side
+        const no_move_start = opts.no_move_start ?? move_start + 50;
+        const no_move_end = move_end - 50;
+        verify_no_move_range(no_move_start, no_move_end);
+    }
+
     reset_list({count: 51});
-    verify_no_move_range(0, 51);
+    verify_no_move_range(0, 51); // This is the whole list
 
     reset_list({count: 450});
+    // 400 messages rendered, with the last 50 in the move range
     verify_no_move_range(0, 350);
 
-    verify_move(350, [150, 450]);
-    verify_no_move_range(200, 400);
+    verify_move_and_no_move_range(350, {
+        // top maxes out at 450
+        move_end: 450,
+    });
 
-    verify_move(199, [0, 400]);
-    verify_no_move_range(50, 350);
+    // We load more than 200 on the upper end, because we load the full 400
+    // messages and 199 is less than half of that.
+    verify_move_and_no_move_range(199, {
+        move_start: 0,
+        move_end: 400,
+    });
 
-    verify_move(350, [150, 450]);
-    verify_no_move_range(200, 400);
+    verify_move_and_no_move_range(350, {
+        // top maxes out at 450
+        move_end: 450,
+    });
 
-    verify_move(199, [0, 400]);
-    verify_no_move_range(0, 350);
+    verify_move_and_no_move_range(199, {
```

Comment:
I'm not really sure why this case was also here, since it's the same as the 199 case above. The above case could (and probably should) also have the `no_move_start: 0`

Reply:
I'm not sure either; maybe the answer is that the first case was intended to have a wider window?

Reply:
OK I read this more carefully and figured out what's going on. I've now added some comments that explain the test a bit better; the key point is that it's a sequence of jumps and we're confirming the updates are correct for each of them.